# Confidence Intervals

In [56]:
import numpy as np
from scipy import stats

In [57]:
def se(hatpi, n):
    return np.sqrt((hatpi*(1-hatpi))/n)

In [58]:
def ci(hatpi, se, z):
    pm = z*se
    print("π ± z(se) = {0} ± {1}({2:2.3f}) = {3} ± {4:2.4f}, or ({5:2.4f},{6:2.4f})".format(hatpi,z,se, hatpi,pm, hatpi-pm,hatpi+pm))

In [65]:
hatpi = 0.538
n = 1095
z = 2.58

# for rounding, below
sigdig = 3

In [66]:
se(hatpi, n)

0.015066246407845513

In [67]:
ci(hatpi, round(se(hatpi, n),sigdig), z)

π ± z(se) = 0.538 ± 2.58(0.015) = 0.538 ± 0.0387, or (0.4993,0.5767)


## Problem
In response to the GSS question in 2014 about the number of hours daily spent watching TV, the responses by the seven subjects who identified themselves as being raised Islamic were 1, 1, 1, 2, 2, 3, 6

### Estimate the mean, standard deviation, and standard error

In [137]:
data = [1, 1, 1, 2, 2, 3, 6]
n = len(data)

#### the mean

In [138]:
mu = np.mean(data)

In [139]:
mu

2.2857142857142856

#### the standard deviation

In [140]:
s = np.std(data, ddof=1)

In [141]:
s

1.7994708216848745

#### estimated standard error
$se = \frac{s}{\sqrt{n}}$

In [142]:
se = s / np.sqrt(n)

In [143]:
se

0.6801360408136047

verify using Scipy's `stats` package

In [144]:
stats.sem(data)

0.6801360408136047

#### summary

In [145]:
precision = 3

In [146]:
print("data:     {}".format(data))
print("n:        {}".format(n))
print("Mean:     {}".format(round(mu, precision)))
print("StDev:    {}".format(round(s, precision)))
print("SE Mean:  {}".format(round(se, precision)))

data:     [1, 1, 1, 2, 2, 3, 6]
n:        7
Mean:     2.286
StDev:    1.799
SE Mean:  0.68


#### confidence interval, using t-score
$\bar{y} \pm t_{.025}(se)$, with $se = s / \sqrt{n}$, where $df = n-1$ for the t-score. In this case $df = 7-1 = 6$.

$\bar{y} \pm t_{.025}(se)$   
$\bar{y} \pm t_{.025}(s / \sqrt{n})$   
$2.286 \pm 2.447(1.799/\sqrt{7})$

In [147]:
pm = 2.447*(s/np.sqrt(n))

In [148]:
pm

1.6642928918708908

In [149]:
print("{} +/- {}".format(round(mu,3), round(pm,3)))

2.286 +/- 1.664


In [150]:
low = mu - pm

In [151]:
high = mu + pm

In [152]:
print("({}, {})".format(round(low,3), round(high,3)))

(0.621, 3.95)
